src/wp-includes/blocks/cover.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Server-side rendering of the `core/cover` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/cover` block on server.
+ *
+ * @param array $attributes The block attributes.
+ * @param array $content    The block rendered content.
+ *
+ * @return string Returns the cover block markup, if useFeaturedImage is true.
+ */
+function render_block_core_cover( $attributes, $content ) {
+	if ( false === $attributes['useFeaturedImage'] ) {
+		return $content;
+	}
+
+	$current_featured_image = get_the_post_thumbnail_url();
+
+	if ( false === $current_featured_image ) {
+		return $content;
+	}
+
+	$is_img_element      = ! ( $attributes['hasParallax'] || $attributes['isRepeated'] );
+	$is_image_background = 'image' === $attributes['backgroundType'];
+
+	if ( $is_image_background && ! $is_img_element ) {
+		$content = preg_replace(
+			'/class=\".*?\"/',
+			'${0} style="background-image:url(' . esc_url( $current_featured_image ) . ')"',
+			$content,
+			1
+		);
+	}
+
+	if ( $is_image_background && $is_img_element ) {
+		$object_position = '';
+		if ( isset( $attributes['focalPoint'] ) ) {
+			$object_position = round( $attributes['focalPoint']['x'] * 100 ) . '%' . ' ' .
+			round( $attributes['focalPoint']['x'] * 100 ) . '%';
+		}
+
+		$image_template = '<img
+			class="wp-block-cover__image-background"
+			alt="%s"
+			src="%s"
+			style="object-position: %s"
+			data-object-fit="cover"
+			data-object-position="%s"
+		/>';
+
+		$image = sprintf(
+			$image_template,
+			esc_attr( get_the_post_thumbnail_caption() ),
+			esc_url( $current_featured_image ),
+			esc_attr( $object_position ),
+			esc_attr( $object_position )
+		);
+
+		$content = str_replace(
+			'</span><div',
+			'</span>' . $image . '<div',
+			$content
+		);
+
+	}
+
+	return $content;
+}
+
+/**
+ * Registers the `core/cover` block renderer on server.
+ */
+function register_block_core_cover() {
+	register_block_type_from_metadata(
```

Comment:
This generates an error in the PHPUnit tests as a `_doing_it_wrong()` is thrown because the `core/cover` block is already registered.

This suggestion causes the tests to pass and the featured image still appears on the frontend:
```suggestion
	if ( WP_Block_Type_Registry::get_instance()->is_registered( 'core/cover' ) ) {
		return;
	}

	register_block_type_from_metadata(
```

Reply:
Thanks for testing, @costdev. That's odd since no other core plugins do the check, so maybe I missed something when moving the non-dynamic block to dynamic.

Let's wait for what @gziolo has to say :)

Reply:
We need to unregister the init callback so it doesn’t get executed multiple times for unit tests. It’s super annoying but it’s how all dynamic blocks were handled before in the context of unit tests.

Reply:
Updated in [924d4be](https://github.com/WordPress/wordpress-develop/pull/2601/commits/924d4bec771175f076e7961f58861bc93762cc20).